aten/src/ATen/native/mps/operations/Convolution.mm

```diff
@@ -116,6 +116,66 @@ static void fill_conv_desc(MPSGraphConvolution2DOpDescriptor* descriptor_,
   descriptor_.groups = groups;
 }
 
+static MPSGraphTensor* permuteTensor(MPSGraph* graph, MPSGraphTensor* inputTensor, NSArray* permuteOrder) {
+    NSUInteger srcRank = [[inputTensor shape] count];
+    if (srcRank != [permuteOrder count]) {
+        return nil;
+    }
+    MPSGraphTensor* outputTensor = inputTensor;
+    outputTensor = [graph transposeTensor:outputTensor permutation:permuteOrder name:nil];
+}
+
+static MPSGraphTensor* reshapePermuteReshape(MPSGraph* mpsGraph, MPSGraphTensor* tensor__, MPSShape* reshape1, MPSShape* permutation, MPSShape* reshape2) {
+    MPSGraphTensor *tensor_ = [mpsGraph reshapeTensor:tensor__ withShape:reshape1 name:nil];
+    MPSGraphTensor *tensor;
+    if (@available(macOS 13.0, *)) {
```

Comment:
Don't use `@available`(as it false false if executed from Python runtime built for older MacOS), but rather `is_macos_13_or_newer()`